core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockResolver.scala

```diff
@@ -104,11 +106,12 @@ private[spark] class FileShuffleBlockResolver(conf: SparkConf)
    * Get a ShuffleWriterGroup for the given map task, which will register it as complete
    * when the writers are closed successfully
    */
-  def forMapTask(shuffleId: Int, mapId: Int, numBuckets: Int, serializer: Serializer,
-      writeMetrics: ShuffleWriteMetrics): ShuffleWriterGroup = {
+  def forMapTask(shuffleId: Int, mapId: Int, stageAttemptId: Int, numBuckets: Int,
+      serializer: Serializer, writeMetrics: ShuffleWriteMetrics): ShuffleWriterGroup = {
```

Comment:
To emphasize the tightly-coupled nature of shuffleId and stageAttemptId, I guess I would be inclined to make this method use a ShuffleIdAndAttempt instead of two separate Ints.  I'm also not a huge fan of the ShuffleIdAndAttempt name, but I haven't got a suggestion for a better one at the moment.